files.py

```diff
@@ -157,20 +164,21 @@ def filter_file(file, supabase, vector_store):
     if file.size < 1:
         st.write(f"💨 {file.name} is empty.")
         return False
-
-    if file_extension in file_processors:
-        if st.secrets.self_hosted == "false":
-            file_processors[file_extension](vector_store, file, stats_db=supabase)
+    else:
```

Comment:
Is there a reason you added this `else` condition? Unless I'm missing something it shouldn't be needed since the above conditional statements have return methods in them. It's best to reduce the level of nesting for readability.